test/query.test.ts

```diff
@@ -335,6 +336,23 @@ describe('query', () => {
        assert.equal(str, 'bin(a,q)');
     });
 
+
+    it('should return correct fieldDef string for bin field with maxbins', () => {
+       const str = stringifyEncodingQueryFieldDef({
+         channel: Channel.X, field: 'a', type: Type.QUANTITATIVE, bin: {maxbins: 20}
+       });
+       assert.equal(str, 'bin(a,q,maxbins=20)');
+    });
+
+
+    it('should return correct fieldDef string for scale with scaleType log', () => {
+       const str = stringifyEncodingQueryFieldDef({
+         channel: Channel.X, field: 'a', type: Type.QUANTITATIVE, scale: {type: ScaleType.LOG}
+       });
+       assert.equal(str, 'scale(a,q,scale={type: log})');
```

Comment:
This is incorrect.   scale isn't a field transformation function.  

This should be `a,q,scale={type: log}`.  


Also add another case when scale.type is specified for another function.  
e.g.,  having `bin` + `scale.type` (even better include `bin.maxbins`)